diff --git a/refind/main.c b/refind/main.c
index 1efadf6..be4c431 100644
--- a/refind/main.c
+++ b/refind/main.c
@@ -142,8 +142,8 @@ static REFIT_MENU_SCREEN AboutMenu      = { L"About", NULL, 0, NULL, 0, NULL, 0,
 REFIT_CONFIG GlobalConfig = { FALSE, TRUE, FALSE, 0, 0, 0, DONT_CHANGE_TEXT_MODE, 20, 0, 0, GRAPHICS_FOR_OSX, LEGACY_TYPE_MAC, 0, 0,
                               { DEFAULT_BIG_ICON_SIZE / 4, DEFAULT_SMALL_ICON_SIZE, DEFAULT_BIG_ICON_SIZE }, BANNER_NOSCALE,
                               NULL, NULL, CONFIG_FILE_NAME, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL,
-                              { TAG_SHELL, TAG_MEMTEST, TAG_GDISK, TAG_APPLE_RECOVERY, TAG_WINDOWS_RECOVERY, TAG_MOK_TOOL,
-                                TAG_ABOUT, TAG_SHUTDOWN, TAG_REBOOT, TAG_FIRMWARE, 0, 0, 0, 0, 0, 0 }
+                              { TAG_SHELL, TAG_MEMTEST, TAG_GDISK, TAG_KITLUGIA, TAG_APPLE_RECOVERY, TAG_WINDOWS_RECOVERY,
+                                TAG_MOK_TOOL, TAG_ABOUT, TAG_SHUTDOWN, TAG_REBOOT, TAG_FIRMWARE, 0, 0, 0, 0, 0, 0 }
                             };
 
 EFI_GUID GlobalGuid = EFI_GLOBAL_VARIABLE;
@@ -2343,16 +2343,26 @@ static VOID ScanForTools(VOID) {
             break;
 
          case TAG_GDISK:
-            j = 0;
-            while ((FileName = FindCommaDelimited(GDISK_NAMES, j++)) != NULL) {
-               if (FileExists(SelfRootDir, FileName)) {
-                  AddToolEntry(SelfLoadedImage->DeviceHandle, FileName, L"disk partitioning tool",
-                               BuiltinIcon(BUILTIN_ICON_TOOL_PART), 'G', FALSE);
-               } // if
-               MyFreePool(FileName);
-            } // while
-            FileName = NULL;
-            break;
+             j = 0;
+             while ((FileName = FindCommaDelimited(GDISK_NAMES, j++)) != NULL) {
+                if (FileExists(SelfRootDir, FileName)) {
+                   AddToolEntry(SelfLoadedImage->DeviceHandle, FileName, L"disk partitioning tool",
+                                BuiltinIcon(BUILTIN_ICON_TOOL_PART), 'G', FALSE);
+                } // if
+                MyFreePool(FileName);
+             } // while
+             FileName = NULL;
+             break;
+
+         case TAG_KITLUGIA:
+             FileName = StrDuplicate(L"\\EFI\\KitLugia\\kitlugia_shrink.efi");
+             if (FileExists(SelfRootDir, FileName)) {
+                AddToolEntry(SelfLoadedImage->DeviceHandle, FileName, L"KitLugia Recovery",
+                             BuiltinIcon(BUILTIN_ICON_TOOL_KITLUGIA), 'K', FALSE);
+             }
+             MyFreePool(FileName);
+             FileName = NULL;
+             break;
 
          case TAG_APPLE_RECOVERY:
             FileName = StrDuplicate(L"\\com.apple.recovery.boot\\boot.efi");
@@ -2617,6 +2627,10 @@ efi_main (EFI_HANDLE ImageHandle, EFI_SYSTEM_TABLE *SystemTable)
                 RebootIntoFirmware();
                 break;
 
+            case TAG_KITLUGIA: // KitLugia Recovery
+                StartTool((LOADER_ENTRY *)ChosenEntry);
+                break;
+
         } // switch()
     } // while()
 
