diff --git a/refind/icns.h b/refind/icns.h
index 25b545e..1fd1b66 100644
--- a/refind/icns.h
+++ b/refind/icns.h
@@ -67,10 +67,11 @@ EG_IMAGE * BuiltinIcon(IN UINTN Id);
 #define BUILTIN_ICON_TOOL_WINDOWS_RESCUE   (9)
 #define BUILTIN_ICON_TOOL_MOK_TOOL         (10)
 #define BUILTIN_ICON_TOOL_MEMTEST          (11)
-#define BUILTIN_ICON_VOL_INTERNAL          (12)
-#define BUILTIN_ICON_VOL_EXTERNAL          (13)
-#define BUILTIN_ICON_VOL_OPTICAL           (14)
-#define BUILTIN_ICON_COUNT                 (15)
+#define BUILTIN_ICON_TOOL_KITLUGIA         (12)
+#define BUILTIN_ICON_VOL_INTERNAL          (13)
+#define BUILTIN_ICON_VOL_EXTERNAL          (14)
+#define BUILTIN_ICON_VOL_OPTICAL           (15)
+#define BUILTIN_ICON_COUNT                 (16)
 
 #endif
 
